### DISCUSSION TITLE
ESP32 Automatically Build a Thread Network

### DISCUSSION THREAD

**Discussion by im22361** [openthread/openthread#9922] [Q&A] [answered]
Hello, 

I am new to ESP32 and OpenThread.
I have followed the steps by OpenThread  about how to Build a Thread Network with the ESP32H2 and ESP Thread Border Router Board 
(https://openthread.io/codelabs/esp-openthread-hardware#3). 
 I was able to connect my ESP Thread Border Router to my ESP32C6. However,   I wanted to know if it was possible to automate this process and not have to type the commands in the CLI everytime. 
Is there a variable that needs to be set?  or would I need to run a script to go through the  Thread CLI command?
I would like to be able to power up my ESP32s and have them automatically connect to each other.
Any support would be greatly appreciated

**Comment by jwhui** [comment]
You can modify the [application code](https://github.com/openthread/openthread/blob/main/examples/apps/cli/main.c) to call the appropriate OT APIs for starting Thread. For example, if an Active Operational Dataset exists, call `otIp6SetEnabled()` and `otThreadSetEnabled()`.

**Reply by im22361** [reply]
Thank you for this. 
I added those functions and now the ESP32C6 automatically starts up, however it is always a Leader rather than a child or router. I have to then manually type: ifconfig down, ifconfig up and thread start, so that it is Child. 

Can you please advise ?

**Reply by jwhui** [reply]
The `ifconfig` and `thread` commands simply call the same APIs. You might need to adjust when those APIs are called. 

https://github.com/openthread/openthread/blob/3ebc3d3a06177ab165fb2eb6f3b28f8f53a65a53/src/cli/cli.cpp#L3159

https://github.com/openthread/openthread/blob/3ebc3d3a06177ab165fb2eb6f3b28f8f53a65a53/src/cli/cli.cpp#L6972

**Reply by ghost** [reply]
can you help me to automate the thread border router? I couldn't find a possible method and having trouble with it, do I want to edit the ot_rcp code or esp_ot_br code? i am using esp32s3 and esp32h2 router board. i have automated the ot_cli code and it works perfectly. Any support would be greatly appreciated.

**Reply by chshu** [reply, marked as answer]
Please follow this instruction to enable the auto start mode: https://docs.espressif.com/projects/esp-thread-br/en/latest/dev-guide/build_and_run.html#wi-fi-based-thread-border-router, it will join the pre-configured Wi-Fi network and form a Thread network automatically.
